src/main/java/rxbroadcast/VectorTimestamp.java

```diff
@@ -76,26 +78,30 @@ public boolean equals(final Object o) {
 
     @Override
     public int hashCode() {
-        return Objects.hash(new Object() {
-            @Override
-            public boolean equals(final Object o) {
-                return this == o;
-            }
-
-            @Override
-            public int hashCode() {
-                return Arrays.hashCode(ids);
-            }
-        }, new Object() {
-            @Override
-            public boolean equals(final Object o) {
-                return this == o;
-            }
-
-            @Override
-            public int hashCode() {
-                return Arrays.hashCode(timestamps);
-            }
-        });
+        if (hashCode == null) {
+            hashCode = Objects.hash(new Object() {
```

Comment:
This field may need to be calculated in the constructor to be able to automate testing of the equals contract